Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -0,0 +1,420 @@
+        """
+        self.server = None
+        self.api_root = None
+        self.collections = None
+
+        self.collection_to_fetch = collection_to_fetch
+        self.base_url = url
+        self.proxies = proxies
+        self.verify = verify
+        self.auth = None
+        if username and password:
+            if username == DFLT_API_USERNAME:
+                self.auth = TokenAuth(key=password)
+            else:
+                self.auth = requests.auth.HTTPBasicAuth(username, password)
+
+        self.field_map = field_map if field_map else {}
+        self.indicator_regexes = [
+            re.compile(INDICATOR_EQUALS_VAL_PATTERN),
+            re.compile(HASHES_EQUALS_VAL_PATTERN),
+        ]
+        self.cidr_regexes = [
+            re.compile(CIDR_ISSUBSET_VAL_PATTERN),
+            re.compile(CIDR_ISUPPERSET_VAL_PATTERN),
+        ]
+
+    def init_server(self, version=TAXII_VER_2_0):
+        """
+        Initializes a server in the requested version
+        :param version: taxii version key (either 2.0 or 2.1)
+        """
+        server_url = urljoin(self.base_url)
+        if version is TAXII_VER_2_0:
+            self.server = v20.Server(
+                server_url, verify=self.verify, auth=self.auth, proxies=self.proxies,
+            )
+        else:
+            self.server = v21.Server(
+                server_url, verify=self.verify, auth=self.auth, proxies=self.proxies,
+            )
+
+    def init_roots(self):
+        """
+        Initializes the api roots (used to get taxii server objects)
+        """
+        if not self.server:
+            raise ValueError(
+                "Please init server prior to calling init_roots. Cannot init roots without server."
+            )
+        try:
+            # try TAXII 2.0
+            self.api_root = self.server.api_roots[0]
+        except (TAXIIServiceException, HTTPError) as e:
+            if isinstance(e, HTTPError) and "406 Client Error" not in str(e):
+                raise e
+            # switch to TAXII 2.1
+            self.init_server(version=TAXII_VER_2_1)
+            self.api_root = self.server.api_roots[0]
+
+    def init_collections(self):
+        """
+        Collects available taxii collections
+        """
+        self.collections = [x for x in self.api_root.collections]  # type: ignore[union-attr]
+
+    def init_collection_to_fetch(self, collection_to_fetch=None):
+        """
+        Tries to initialize `collection_to_fetch` if possible
+        """
+        if collection_to_fetch is None and isinstance(self.collection_to_fetch, str):
+            # self.collection_to_fetch will be changed from str -> Union[v20.Collection, v21.Collection]
+            collection_to_fetch = self.collection_to_fetch
+        if not self.collections:
+            raise DemistoException(
+                "No collection is available for this user, please make sure you entered the configuration correctly"
+            )
+        if collection_to_fetch:
```

Comment:
let's also add a case where the user does not add a collection to fetch, and we scan all the collection in the discovery service.
I do not know how common it is, but unit42 works this way...

Reply:
added in https://github.com/demisto/content/pull/7637/commits/536f7120b75a4cd7ffaa1cfd842a9fb7b249215d